tests/unittest/pywbem/test_subscriptionmanager.py

```diff
@@ -369,3 +371,341 @@ def test_create_not_owned_subscription(self):
+            submgr_id="ValidID",
+            connection_attrs='MOCKSERVER',
+            filter_attrs=None,
+            dest_attrs=dict(server_id="http://localhost:5988",
+                            listener_urls="httpx",
+                            owned=True),
+            subscription_attrs=None,
+            exp_result=dict(server_id="http://localhost:5988")
+        ),
+        ValueError, None, OK
+    ),
+    (
+        "Add valid listener_destination",
+        dict(
+            submgr_id="ValidID",
+            connection_attrs='MOCKSERVER',
+            filter_attrs=None,
+            dest_attrs=dict(server_id="http://localhost:5988",
+                            listener_urls="http://localhost:5000",
+
+                            owned=True),
+            subscription_attrs=None,
+            exp_result=dict(server_id="http://localhost:5988",
+                            listener_count=1)
+        ),
+        ValueError, None, OK
+    ),
+    (
+        "Add multiple valid listener_destinations",
+        dict(
+            submgr_id="ValidID",
+            connection_attrs='MOCKSERVER',
+            filter_attrs=None,
+            dest_attrs=dict(server_id="http://localhost:5988",
+                            listener_urls=["http://localhost:5000",
+                                           "https://localhost:5001"],
+                            owned=True),
+            subscription_attrs=None,
+            exp_result=dict(server_id="http://localhost:5988",
+                            listener_count=2)
+        ),
+        ValueError, None, OK
+    ),
+
+    # Tests that need to be added
+    # Add server_id second time, Should fail
+    # Add server where param is not WBEMServer
+    # add not_owned filters and destinations
+    # NOTE: The following tests will be added after merge with issue #2701
+    # Remove multiple owned subscriptions, filters, dests
+    # NOTE: Because of issue# 2704, we cannot test a number of capabilities
+    # NOTE: Eventually this should replace most of the unit tests.
+]
+
+
+@pytest.mark.parametrize(
+    "desc, kwargs, exp_exc_types, exp_warn_types, condition",
+    TESTCASES_SUBMGR)
+@simplified_test_function
+def test_submgr(testcase, submgr_id, connection_attrs, filter_attrs,
+                dest_attrs, subscription_attrs, exp_result):
+    """
+    Tests object init and add-server
+    """
+    # The code to be tested.
+    submgr = WBEMSubscriptionManager(submgr_id)
+    # Tests for add_server
+    server_id = None
+    server_ids = []
+
+    def define_connection(connection_attrs):
+        """
+        Define connection, either 'MOCKSERVER' or WBEMConnection and return
+        the resulting server_id.
+
+        Returns:
+            returns the server_id
+        """
+
+        # setup a WBEMConnection or pywbem_mock server
+        if connection_attrs == 'MOCKSERVER':
+            conn = setup_mock_environment()
+        else:
+            conn = WBEMConnection(**connection_attrs)
```

Comment:
This is supposed to be a unit test. In a unit test, we should not connect to a real server. That should be done in an end2end test, where we have our server selection mechanisms.

Reply:
The only tests we do with a real WBEMConnection will be failure tests.

Reply:
That's fine, but that should be done in end2end tests and not in unit tests.